cdk/downloader_stack.py

```diff
@@ -286,6 +286,59 @@ def __init__(
             threshold=1,
         )
 
+        link_subscription = aws_lambda_python.PythonFunction(
+            self,
+            id=f"{identifier}-link-subscription",
+            entry="lambdas/link_fetcher",
+            index="app/subscription_handler.py",
+            handler="handler",
+            layers=[
+                db_layer,
+            ],
+            memory_size=200,
+            timeout=Duration.minutes(15),
+            runtime=aws_lambda.Runtime.PYTHON_3_11,
```

Comment:
+1 it looks specific to the Python-specific `aws_lambda_python_alpha` version of CDK. I was wondering how you got the GIBS stuff working on Py3.12!

Specifically this Python-specific Lambda library installs Pipenv and Poetry as part of setup, and there's some `pipenv` <> `py3.12` interaction that is going badly, see https://github.com/aws/aws-cdk/issues/30170#issuecomment-2137950708

They're tracking the issue but there isn't a resolution yet. We could avoid using `aws_lambda_python_alpha` if we want to export the `Pipfile.lock` to a `requirements.txt` ahead of the CDK deploy stage, but I'm hoping there'll be a fix upstream instead of taking on that work